moveit_plugins/moveit_simple_controller_manager/src/follow_joint_trajectory_controller_handle.cpp

```diff
@@ -54,34 +54,42 @@ bool FollowJointTrajectoryControllerHandle::sendTrajectory(const moveit_msgs::ms
   }
 
   if (done_)
-    RCLCPP_DEBUG_STREAM(LOGGER, "sending trajectory to " << name_);
+    RCLCPP_INFO_STREAM(LOGGER, "sending trajectory to " << name_);
   else
-    RCLCPP_DEBUG_STREAM(LOGGER, "sending continuation for the currently executed trajectory to " << name_);
+    RCLCPP_INFO_STREAM(LOGGER, "sending continuation for the currently executed trajectory to " << name_);
 
   control_msgs::action::FollowJointTrajectory::Goal goal = goal_template_;
   goal.trajectory = trajectory.joint_trajectory;
 
   rclcpp_action::Client<control_msgs::action::FollowJointTrajectory>::SendGoalOptions send_goal_options;
   // Active callback
-  send_goal_options.goal_response_callback = [this](const auto& /* unused-arg */) {
-    RCLCPP_DEBUG_STREAM(LOGGER, name_ << " started execution");
+  send_goal_options.goal_response_callback = [this](const auto& future) {
+    RCLCPP_INFO_STREAM(LOGGER, name_ << " started execution");
+    auto goal_handle = future.get();
+    if (!goal_handle)
+      RCLCPP_INFO(LOGGER, "Goal request rejected");
+    else
+      RCLCPP_INFO(LOGGER, "Goal request accepted!");
   };
   // Result callback
   send_goal_options.result_callback =
       std::bind(&FollowJointTrajectoryControllerHandle::controllerDoneCallback, this, _1);
   // Send goal
   auto current_goal_future = controller_action_client_->async_send_goal(goal, send_goal_options);
-  if (rclcpp::spin_until_future_complete(node_, current_goal_future) != rclcpp::executor::FutureReturnCode::SUCCESS)
+  std::future_status status = current_goal_future.wait_for(std::chrono::seconds(60));
```

Comment:
Why wait for 60 seconds?  the goal doesn't really have a timeout, right?

Reply:
i was thinking it would be better to have a timeout instead of spinning forever.  Besides https://github.com/ros-planning/moveit/blob/master/moveit_plugins/moveit_simple_controller_manager/src/follow_joint_trajectory_controller_handle.cpp#L65 seems to be completely async with no checking. Should we follow it as well?

Reply:
I think we can keep this spinning since it's the job of the TrajectoryExecutionMonitor to validate execution duration (and cancel the goal if execution times out). I wouldn't set the goal to a hard timeout at least.

Reply:
removed timeout 465027a